include/deal.II/physics/physics.h

```diff
@@ -0,0 +1,137 @@
+ * definition, construction and manipulation of data related to
+ * physical fields and materials.
+ *
+ * @ingroup Physics
+ */
+namespace Physics
+{
+
+  /**
+   * This namespace provides a collection of definitions that
+   * conform to standard notation used in (nonlinear) elasticity.
+   *
+   * <h3>Notation</h3>
+   *
+   * References for this notation include:
+   * @code{.bib}
+       @Book{Holzapfel2001a,
+          title =     {Nonlinear solid mechanics. A Continuum Approach for Engineering},
+          publisher = {John Wiley \& Sons Ltd.},
+          year =      {2007},
+          author =    {Holzapfel, G. A.},
+          address =   {West Sussex, England},
+          note =      {ISBN: 0-471-82304-X}
+        }
+        @Book{Wriggers2008a,
+          title =     {Nonlinear finite element methods},
+          publisher = {Springer Berlin Heidelberg},
+          year =      {2008},
+          author =    {Wriggers, P.},
+          volume =    {4},
+          address =   {Berlin, Germany},
+          note =      {ISBN: 978-3-540-71000-4},
+          doi =       {10.1007/978-3-540-71001-1}
+        }
+   * @endcode
+   *
+   * For convenience we will predefine some commonly referenced tensors and
+   * operations.
+   * Considering the position vector $\mathbf{X}$ in the referential (material)
+   * configuration, points $\mathbf{X}$ are transformed to points $\mathbf{x}$
+   * in the current (spatial) configuration through the nonlinear map
+   * @f[
+   *  \mathbf{x}
+   *   := \boldsymbol{\varphi} \left( \mathbf{X} \right)
+   *    = \mathbf{X} + \mathbf{u}(\mathbf{X}) \, ,
+   * @f]
+   * where the $\mathbf{u}(\mathbf{X})$ represents the displacement vector.
+   * From this we can compute the deformation gradient tensor as
+   * @f[
+   *  \mathbf{F} := \mathbf{I} + \nabla_{0}\mathbf{u} \, ,
+   * @f]
+   * wherein the differential operator $\nabla_{0}$ is defined as
+   * $\frac{\partial}{\partial \mathbf{X}}$ and $\mathbf{I}$ is the identity
+   * tensor.
+   *
+   * Finally, two common tensor operators are represented by $\cdot$ and $:$
+   * operators. These respectively represent a single and double contraction over
+   * the inner tensor indices.
+   *
+   * One can think of fourth-order tensors as linear operators mapping second-order
+   * tensors (matrices) onto themselves in much the same way as matrices map
+   * vectors onto vectors.
+   * To provide some context to the implemented class members and functions,
+   * consider the following fundamental operations performed on tensors with special
+   * properties:
+   *
+   * If we represent a general second-order tensor as $\mathbf{A}$, then the general
+   * fourth-order unit tensors $\mathcal{I}$ and $\overline{\mathcal{I}}$ are
+   * defined by
+   * @f[
+   *  \mathbf{A} = \mathcal{I}:\mathbf{A}
+   *        \qquad \text{and} \qquad
+   *      \mathbf{A}^T = \overline{\mathcal{I}}:\mathbf{A} \, .
+   * @f]
+   * Note that $\mathcal{I} \neq \overline{\mathcal{I}}^T$.
```

Comment:
Would it be worthwhile pointing out that in index notation, you have
```
  I_{ijkl} = \delta_{ik}\delta_{jl}
  \overline I_{ijkl} = \delta_{il}\delta_{jk}
```
with Kronecker deltas as commonly defined.